dbms/src/Storages/DeltaMerge/SSTFilesToDTFilesOutputStream.cpp

```diff
@@ -289,6 +307,42 @@ void SSTFilesToDTFilesOutputStream<ChildStream>::cancel()
     }
 }
 
+template <typename ChildStream>
+void SSTFilesToDTFilesOutputStream<ChildStream>::updateRangeFromNonEmptyBlock(Block & block)
+{
+    // TODO: Now we update the rangeEnd for every block. We can update the rangeEnd only once,
+    //  just before a new DTFile is going to be created.
+
+    RUNTIME_CHECK(block.rows() > 0);
+
+    const auto & pk_col = block.getByName(MutableSupport::tidb_pk_column_name);
+    const auto rowkey_column = RowKeyColumnContainer(pk_col.column, schema_snap->is_common_handle);
+    auto & current_file_range = ingest_files_range.back();
+
+    auto const block_start = rowkey_column.getRowKeyValue(0);
+    auto const block_end = rowkey_column.getRowKeyValue(pk_col.column->size() - 1) //
+                               .toRowKeyValue()
+                               .toPrefixNext(); // because range is right-open.
+
+    // TODO: what if there are a lof of versions, exceeding a block size? Will our checks fail?
+    RUNTIME_CHECK(compare(block_start, block_end.toRowKeyValueRef()) < 0);
+
+    if (!current_file_range.has_value())
+    {
+        current_file_range = RowKeyRange( //
+            block_start.toRowKeyValue(),
+            block_end,
+            schema_snap->is_common_handle,
+            schema_snap->rowkey_column_size);
+    }
+    else
+    {
+        RUNTIME_CHECK(compare(block_start, current_file_range->getStart()) > 0);
```

Comment:
@JaySon-Huang Will we split a key into multiple blocks (due to multiple versions)? If so, then this check may fail I think.

Reply:
Never have overlapped ranges, the children `BoundedSSTFilesToBlockInputStreamPtr` ensure that for the correctness of rough set filter in DTFile

Reply:
Good news. I will add some checks in the next PR along with the test.